src/wp-includes/rest-api/endpoints/class-wp-rest-settings-controller.php

```diff
@@ -77,10 +92,15 @@ public function get_item_permissions_check( $request ) {
 	 * @return array|WP_Error Array on success, or WP_Error object on failure.
 	 */
 	public function get_item( $request ) {
-		$options  = $this->get_registered_options();
-		$response = array();
+		$options            = $this->get_registered_options();
+		$response           = array();
+		$can_manage_options = current_user_can( 'manage_options' );
 
 		foreach ( $options as $name => $args ) {
+			if ( ! $can_manage_options && ! $args['schema']['public'] ) {
+				continue;
```

Comment:
So requests from users with `manage_options` will have the `rest_pre_get_setting` filter but other requests will not? I think it would be best if there was a way to ensure that the filter is run for all requests